app/components/provider_interface/training_with_disability_component.rb

```diff
@@ -3,15 +3,34 @@ class TrainingWithDisabilityComponent < ViewComponent::Base
     include ViewHelper
 
     attr_reader :application_form
+    delegate :disclose_disability?, :disability_disclosure, to: :application_form
 
     def initialize(application_form:)
       @application_form = application_form
     end
 
-    def disability_disclosure
-      return application_form.disability_disclosure if application_form.disclose_disability? && application_form.disability_disclosure.present?
+    def rows
+      rows = [{ key: 'Do you want to ask for help to become a teacher?', value: disability_disclosure_support }]
 
-      'No information shared.'
+      if disclose_information?
+        rows << { key: 'Give any relevant information', value: disability_disclosure }
+      end
+
+      rows
+    end
+
+  private
+
+    def disclose_information?
+      disclose_disability? && disability_disclosure.present?
+    end
+
+    def disability_disclosure_support
+      if disclose_information?
+        'Yes, I want to share information about myself so my provider can take steps to support me'
```

Comment:
Is it worth referencing the text from the locale file for candidates here ([training_with_a_disability.yml#L6](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/config/locales/candidate_interface/training_with_a_disability.yml#L6)) rather than hardcoding it, so that it can stay in sync if we update wording in the candidate interface?

Reply:
Sounds good. I am happy to do it :)

Reply:
Done ♻️

Reply:
Thanks! 🙌 

Get a quick code review from someone and then we can ship this?